src/components/Button/index.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { PropsWithChildren, MouseEvent } from 'react';
+import { styButton } from './styles';
+import { ButtonProps } from './buttonProps';
+import isRoutes from '../../utils/isRoutes';
+
+function cssClassName(btnSize: string, btnType: string): string {
+    switch (btnType) {
+        case 'primary':
+            return `${btnSize} ${styButton.Primary}`;
+        case 'secondary':
+            return `${btnSize} ${styButton.Secondary}`;
+        case 'link':
+            return `${btnSize} ${styButton.Link}`;
+        default:
+            return 'wrong button type';
+    }
+}
+
+const Button = (props: PropsWithChildren<ButtonProps>): JSX.Element => {
+    const {
+        btnSize = 'medium',
+        btnType = 'primary',
+        btnColor,
+        link,
+        element,
+        children,
+        ...rest
+    } = props;
+    const className = cssClassName(btnSize, btnType);
+
+    const color = { backgroundColor: btnColor };
+
+    const handleClick = (e: MouseEvent): void => {
+        if (link !== undefined) {
+            if (element !== undefined) {
+                if (link.includes('http')) {
+                    isRoutes.isValidHttpUrl(link);
+                }
+                if (e.metaKey || e.ctrlKey) {
+                    return;
+                }
+                e.preventDefault(); // cegah loading
```

Comment:
```suggestion
                // rifian.fernando@gmail.com prevents it from loading
                e.preventDefault();
```

commentnya lebih bagus gini, dipisah line jadi kena blame di git